### PR TITLE
[ADVAPP-786] Introduce Draft with AI for knowledge base articles

### PR DESCRIPTION
**Pull request by ankit-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-786

### Technical Description

> Draft knowledge base article with AI assistant when editing the article. 

### Any deployment steps required?

> No

### Are any Feature Flags Added?

> No

_______________________________________________

### CODE REVIEW

File: app-modules/knowledge-base/src/Filament/Actions/DraftWithAiAction.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                $userJobTitle = auth()->user()->job_title ?? 'staff member';
+                $clientName = app(LicenseSettings::class)->data->subscription->clientName;
+                $articleTitle = $livewire?->data['title'];
+
+                try {
+                    $content = $service->complete(
+                        <<<EOL
+                                    My name is {$userName}, and I am a {$userJobTitle} at {$clientName}. I am currently editing an article in the knowledge base of the college. The current title is "{$articleTitle}". 
+
+                                    Please provide the title as a plain string, without any prefixes, labels like "Title:", or formatting such as bold, italics, or quotes. The title should appear as a simple string.
```
Comment (Orrison) [inline app-modules/knowledge-base/src/Filament/Actions/DraftWithAiAction.php:97]: You may want to be explicit that the title and content should have a newline separation between them like we do for the Engagement prompt

File: app-modules/knowledge-base/src/Filament/Actions/DraftWithAiAction.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+use AdvisingApp\Ai\Models\AiAssistant;
+use AdvisingApp\Ai\Settings\AiSettings;
+use Filament\Forms\Components\Textarea;
+use Filament\Notifications\Notification;
+use Filament\Forms\Components\Actions\Action;
+use AdvisingApp\Authorization\Enums\LicenseType;
+use AdvisingApp\Ai\Exceptions\MessageResponseException;
+use AdvisingApp\Ai\Settings\AiIntegratedAssistantSettings;
+
+class DraftWithAiAction extends Action
```
Comment (Orrison) [inline app-modules/knowledge-base/src/Filament/Actions/DraftWithAiAction.php:56]: Let's rename this `DraftKnowledgeBaseArticleWithAiAction` and related blade file, etc. so we can differentiate it from others.

**Review by Orrison — CHANGES_REQUESTED** [review]
Looks good. Just a couple of notes.

I would like @danharrin to review as well.

File: app-modules/knowledge-base/src/Filament/Actions/DraftKnowledgeBaseArticleWithAiAction.php [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    {
+        $this->mergeTags = $tags;
+
+        return $this;
+    }
+
+    public function getMergeTags(): array
+    {
+        return $this->evaluate($this->mergeTags);
+    }
```
Comment (danharrin) [inline app-modules/knowledge-base/src/Filament/Actions/DraftKnowledgeBaseArticleWithAiAction.php:150]: Please remove the merge tag methods and the merge tags property from here as they aren't used in this context

**Review by danharrin — CHANGES_REQUESTED** [review]

**Review by Orrison — CHANGES_REQUESTED** [review]
This looks good but let's remove the unused merge tags details as @danharrin mentioned

**Review by Orrison — CHANGES_REQUESTED** [review]
This looks good to me. @danharrin if you can re-review and mark your comments resolved if they are.

**Review by Orrison — APPROVED** [review]

**Review by danharrin — APPROVED** [review]

**Pull request merged**